lib/rules/generator-star-spacing.js

```diff
@@ -126,17 +214,23 @@ module.exports = {
             }
 
             const starToken = getStarToken(node);
-
-            // Only check before when preceded by `function`|`static` keyword
             const prevToken = sourceCode.getTokenBefore(starToken);
+            const nextToken = sourceCode.getTokenAfter(starToken);
 
-            if (prevToken.value === "function" || prevToken.value === "static") {
-                checkSpacing("before", prevToken, starToken);
+            let kind = "named";
+
+            if (node.type === "FunctionExpression" && (node.parent.type === "MethodDefinition" || node.parent.type === "Property")) {
+                kind = prevToken.value === "static" ? "static" : "method";
+            } else if (nextToken.value === "(") {
```

Comment:
I think it would be better to check for anonymous functions by verifying that `node.id` is null, rather than checking that the next token is an opening paren.